test/unit/eval_algorithms/test_qa_accuracy.py

```diff
@@ -417,157 +382,24 @@ class TestCaseQAAccuracyEvaluate(NamedTuple):
-        generate_model_predict_response_for_dataset.return_value = test_case.input_dataset_with_generated_model_output
+        mock_get_dataset.return_value = test_case.input_dataset
         eval_algorithm = QAAccuracy(config)
         actual_response = eval_algorithm.evaluate(
-            model=model, dataset_config=test_case.dataset_config, prompt_template=test_case.prompt_template, save=True
+            dataset_config=test_case.dataset_config,
+            save=True,
         )
         assert actual_response == test_case.expected_response
-        assert save_dataset.called
-
-    @pytest.mark.parametrize(
-        "test_case",
-        [
-            TestCaseQAAccuracyEvaluate(
-                input_dataset=QA_DATASET,
-                prompt_template=None,
-                dataset_config=DataConfig(
-                    dataset_name="my_custom_dataset",
-                    dataset_uri="tba",
-                    dataset_mime_type=MIME_TYPE_JSON,
-                    model_input_location="tba",
-                    target_output_location="tba",
-                    model_output_location=None,
-                    category_location="tba",
-                ),
-                input_dataset_with_generated_model_output=None,
-                expected_response=[
-                    EvalOutput(
-                        eval_name="qa_accuracy",
-                        dataset_name="my_custom_dataset",
-                        prompt_template=None,
-                        dataset_scores=DATASET_SCORES,
-                        category_scores=CATEGORY_SCORES,
-                        output_path="/tmp/eval_results/qa_accuracy_my_custom_dataset.jsonl",
-                    )
-                ],
-            )
-        ],
-    )
-    @patch("fmeval.eval_algorithms.qa_accuracy.get_dataset")
-    @patch("fmeval.eval_algorithms.qa_accuracy.save_dataset")
-    @patch("fmeval.eval_algorithms.qa_accuracy.generate_model_predict_response_for_dataset")
-    def test_qa_accuracy_evaluate_without_model(
-        self, generate_model_predict_response_for_dataset, save_dataset, get_dataset, test_case, config
-    ):
-        """
-        GIVEN valid inputs i.e. input data config for a dataset with model_outputs,
-            and no request to save records with scores
-        WHEN QAACCURACY.evaluate() is called
-        THEN correct EvalOutput is returned
-        """
-        get_dataset.return_value = test_case.input_dataset
-        generate_model_predict_response_for_dataset.return_value = test_case.input_dataset_with_generated_model_output
-        eval_algorithm = QAAccuracy(config)
-        actual_response = eval_algorithm.evaluate(model=None, dataset_config=test_case.dataset_config)
-        assert not generate_model_predict_response_for_dataset.called
-        assert not save_dataset.called
-        assert actual_response == test_case.expected_response
-
-    class TestCaseQAAccuracyEvaluateInvalid(NamedTuple):
```

Comment:
Validation of `evaluate` parameters is now handled by `evaluate_dataset` function, which has its own unit tests. Hence these are redundant.